### ISSUE TITLE
bug: iOS - playing video from filesystem / support range requests

### ISSUE THREAD

**Issue by sburnicki** [ionic-team/capacitor#5420]
## Bug Report

As already known, video playback from filesystem on iOS makes trouble. iOS 15 makes this problem even worse as it requests more and smaller chunks, so even small videos don't work. The according apple bug with range headers seems to be fixed, so an implementation should be doable by now. See details below.

### Capacitor Version
  @capacitor/core: 3.4.0

### Platform(s)
iOS > 15.0

### Current Behavior
Starting from iOS 15 there are also issues playing medium sized videos (like 10-20MB) from file system. The initial delay is huge and playback is laggy.

### Expected Behavior
The range header should be parsed and respected, so smooth and instant video playback from filesystem is working on all iOS versions that include the according bugfix from apple.

### Code Reproduction
https://github.com/jcesarmobile/no-ranges-bug

### Additional Context
Investigating the issues, I found that there are more/smaller range requests on video playback in iOS 15, probably since video playback was moved to GPU (see [related webkit bug](https://bugs.webkit.org/show_bug.cgi?id=232076)). This causes even small and medium sized video to be requested a few hundred times, since capacitor doesn't support range requests, resulting in initalial playback delay and lagging playback.

I also found the [closed issues here](https://github.com/ionic-team/capacitor/issues/2634), the [webkit bug](https://bugs.webkit.org/show_bug.cgi?id=203302) that caused this and the [demo project](https://github.com/jcesarmobile/no-ranges-bug) from you @jcesarmobile.

However, the bug seems to be closed by now, and executing the demo code on Simulator/iPhone11/iOS 15.2 gives me the following example output: 
```
Optional(["Range": "bytes=2498560-2551807", "X-Playback-Session-Id": "00ECD4FA-B8B3-4B97-A121-DBD633B2EA7E", "Accept": "*/*", "User-Agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 15_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148"])
urlSchemeTask.request.allHTTPHeaderFields Optional(["Accept": "*/*", "Range": "bytes=2551808-2555903", "User-Agent": "Mozilla/5.0 (iPhone; CPU iPhone OS 15_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148", "X-Playback-Session-Id": "00ECD4FA-B8B3-4B97-A121-DBD633B2EA7E"])
```

So it looks like it should be possible now to implement range request on iOS, right?

**Comment by sburnicki** [comment]
I created a working version based on the code reproduction repo, supporting range requests:
https://github.com/sburnicki/no-ranges-bug (forking on github wasn't possible atm, so did it locally)
with this codes, the video plays instantly and smoothly on Simulator/iPhone11/iOS 15.2

Note: The sample code treats all media requests as range requests. The productive implementation should distinguish between actual range requests and those without range header.

**Comment by muuvmuuv** [comment]
This really is a major issue on iOS, our app is Offline first and videos are really terrible in playback. Sometimes they aren't event loading at all. Really strange that this is not a default feature, all these bytes require way too much time to get :D I am happy to give a helping hand if needed!

<img width="1332" alt="Bildschirmfoto 2022-02-16 um 17 08 58" src="https://user-images.githubusercontent.com/11534760/154307671-2436df74-7494-4279-aa99-79ba3b22fea4.png">

**Comment by sburnicki** [comment]
I thought about creating a PR, but I have no idea how to automatically or manually test the code when I copy the new pieces into `capacitor/WebViewAssetHandler.swift`.

**Comment by mcarnoky001** [comment]
I have an app that is recording videos, storing them in a filesystem and after that, the recorded videos are played in a <video> element. So I was able to test it just this way.

**Comment by Hanho-Kim** [comment]
@sburnicki Which file should I change for testing your codes? in node_modules/@capacitor?

**Comment by Hanho-Kim** [comment]
@sburnicki sburnicki's code is working very well in iOS 15 (This should be PR on the project). But in iOS 14, Header does not include "Ranges". Maybe it is related to the https://bugs.webkit.org/show_bug.cgi?id=203302 .

**Comment by jcesarmobile** [comment]
range requests have been implemented in Capacitor 4.0.0 beta 0
https://github.com/ionic-team/capacitor/pull/5659
